.github/workflows /docker-publish.yml

```diff
@@ -0,0 +1,74 @@
+name: Docker
+
+on:
+  push:
+    # Publish `main` as Docker `latest` image.
+    tags: [ 'v*.*.*' ]
+  workflow_dispatch:
+
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  # Run tests.
+  # See also https://docs.docker.com/docker-hub/builds/automated-testing/
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Run tests
+        run: |
+          if [ -f docker-compose.test.yml ]; then
```

Comment:
What does this if do?

Reply:
If docker-compose.test.yml exist then it will use the contents of that specific file, else it will use default.